content/en/docs/Getting-started/samples/build-push.yaml

```diff
@@ -0,0 +1,110 @@
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: build-push
+spec:
+  params:
+  - name: image-reference
+    type: string
+  workspaces:
+  - name: shared-data
+  tasks: 
+  - name: dockerfile
+    taskRef:
+      name: create-dockerfile
+    workspaces:
+    - name: source
+      workspace: shared-data
+  - name: kaniko-build
+    runAfter: ["dockerfile"]
+    taskRef:
+      name: kaniko
+    workspaces:
+    - name: source
+      workspace: shared-data
+    params:
+    - name: IMAGE
+      value: $(params.image-reference)
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: create-dockerfile
+spec:
+  workspaces:
+  - name: source
+  steps:
+  - name: add-dockerfile
+    workingDir: $(workspaces.source.path)
+    image: bash
+    script: |
+      cat <<EOF > Dockerfile
+      FROM alpine:3.16
+      RUN echo "hello world" > hello.log
+      EOF
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: kaniko
+  labels:
+    app.kubernetes.io/version: "0.6"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/categories: Image Build
+    tekton.dev/tags: image-build
+    tekton.dev/displayName: "Build and upload container image using Kaniko"
+    tekton.dev/platforms: "linux/amd64,linux/arm64,linux/ppc64le"
+spec:
+  description: >-
+    This Task builds a simple Dockerfile with kaniko and pushes to a registry.
+    This Task stores the image name and digest as results, allowing Tekton Chains to pick up
+    that an image was built & sign it.
+  params:
+    - name: IMAGE
+      description: Name (reference) of the image to build.
+    - name: DOCKERFILE
+      description: Path to the Dockerfile to build.
+      default: ./Dockerfile
+    - name: CONTEXT
+      description: The build context used by Kaniko.
+      default: ./
+    - name: EXTRA_ARGS
+      type: array
+      default: []
+    - name: BUILDER_IMAGE
+      description: The image on which builds will run (default is v1.5.1)
+      default: gcr.io/kaniko-project/executor:v1.5.1@sha256:c6166717f7fe0b7da44908c986137ecfeab21f31ec3992f6e128fff8a94be8a5
+  workspaces:
+    - name: source
+      description: Holds the context and Dockerfile
+    - name: dockerconfig
+      description: Includes a docker `config.json`
+      optional: true
+      mountPath: /kaniko/.docker
+  results:
+    - name: IMAGE_DIGEST
+      description: Digest of the image just built.
+    - name: IMAGE_URL
+      description: URL of the image just built.
```

Comment:
Could this be updated to use [the most recent version of the Chains type hinting](https://github.com/tektoncd/chains/blob/main/docs/intoto.md#structured-result-type-hinting) (which supports other artifact types besides images as well)

As a bonus it would be great if it could pull source code and use the `-ARTIFACT_INPUTS` syntax as well, but it seems like you're intentionally trying to keep this example a bit simpler 🙏

Reply:
Sorry it took me this long to get back to this, it's been a busy month. Added `image-ARTIFACT_OUTPUTS` to pipeline and updated the commands to fetch the provenance from the PipelineRun annotations instead of the task

Reply:
Thanks @geriom, looks good to me!